src/handlers/dns/DnsStreamHandler.cpp

```diff
@@ -743,6 +760,9 @@ void DnsMetricsManager::process_dnstap(const dnstap::Dnstap &payload)
     // base event
     new_event(stamp);
     // process in the "live" bucket. this will parse the resources if we are deep sampling
+    if (filtered) {
+        live_bucket()->process_filtered();
```

Comment:
shouldn't this have a return after it, or the `process_dnstap` below be in an else { } ? i'm confused why the unit test passes if so.

Reply:
right calling `process_filtered` is good, but then it goes on to call process_dnstap with the payload, which means it would actually count all those metrics and not really filter it right?